src/etl/NFTHelpers.cpp

```diff
@@ -368,3 +372,30 @@ getNFTData(ripple::TxMeta const& txMeta, ripple::STTx const& sttx)
             return {{}, {}};
     }
 }
+
+std::vector<NFTsData>
+getNFTDataFromObj(
+    std::uint32_t const seq,
+    std::string const& key,
+    std::string const& blob)
+{
+    std::vector<NFTsData> nfts;
+    ripple::STLedgerEntry const sle = ripple::STLedgerEntry(
+        ripple::SerialIter{blob.data(), blob.size()},
+        ripple::uint256::fromVoid(key.data()));
```

Comment:
We can probably avoid deserializing the whole object before checking the type. The type is usually encoded somewhere in the first few bytes

Reply:
oh nice - i was bummed about having to deserialize just to do this as well